post-types/ckan-backend-local-harvester.php

```diff
@@ -342,4 +341,142 @@ public static function get_source_type_form_field_options() {
 
 		return $source_type_options;
 	}
+
+	/**
+	 * Applies harvest-status filter
+	 *
+	 * @param WP_Query $query The current query.
+	 */
+	public function filter_posts_by_harvest_status( $query ) {
+		global $post_type, $pagenow;
+
+		if (
+			// Only filter when were on the edit page of ckan-local-harvesters
+			self::POST_TYPE === $post_type &&
+			'edit.php' === $pagenow &&
+			// Only filter when ckan-local-harvesters are queried
+			! empty( $query->query_vars['post_type'] ) &&
+			$query->query_vars['post_type'] === self::POST_TYPE
+		) {
+			$harvest_status_filter   = '';
+			if ( isset( $_GET['harvest_status_filter'] ) ) {
+				$harvest_status_filter = sanitize_text_field( $_GET['harvest_status_filter'] );
+			}
+
+			$harvest_ids_by_state = self::get_harvest_ids_by_status();
+
+			if ( ! empty( $harvest_status_filter ) ) {
```

Comment:
may check if `$harvest_status_filter` in `$harvest_ids_by_state` because this field is read via GET parameters, it's easy to create a PHP error here